testsuite/MDAnalysisTests/lib/test_cutil.py

```diff
@@ -46,6 +48,14 @@ def test_unique_int_1d(values):
     assert res.dtype == ref.dtype
 
 
+def test_array_unique_int_1d():
+    try:
+        res_indices = np.array([6, 5, 5, 7], dtype=np.int_)
+        assert_equal(unique_int_1d(res_indices),  [5, 6, 7])
+    except ValueError:
+        raise AssertionError("Issue #2687 failure")
```

Comment:
I would just let `pytest`/Python handle the exception and use a `# see gh-2687` comment at the start of the test function

Reply:
The reason why I suggested this approach is because if we let the code raise ValueError then the test will fail differently that if it raised AssertionError. Because we know about one specific failure I suggested to make this known failure clear.